### PR TITLE
Added Obsolete to AppDomain.CreateDomain for monodroid. 

### PR DESCRIPTION
**Pull request by steveisok**
The intent is to be helpful for when .NET 5 comes along.

### CODE REVIEW

File: mcs/class/corlib/System/AppDomain.cs [hunk 1]
```diff
@@ -1000,11 +1000,17 @@ internal static String GetProcessGuid ()
 		}
 
 #if MONO_FEATURE_MULTIPLE_APPDOMAINS
+#if MONODROID
+		[Obsolete ("AppDomain.CreateDomain will no longer be supported in .NET 5 and later.  Consider AssemblyLoadContext when it becomes available https://docs.microsoft.com/en-us/dotnet/api/system.runtime.loader.assemblyloadcontext?view=netcore-3.0")]
```
Comment (akoeplinger) [inline mcs/class/corlib/System/AppDomain.cs]: I wonder whether https://docs.microsoft.com/dotnet/core/dependency-loading/understanding-assemblyloadcontext is a better target page

Comment (steveisok) [reply mcs/class/corlib/System/AppDomain.cs]: Agreed

**Comment by akoeplinger** [comment]
@monojenkins commit apidiff

File: mcs/class/corlib/System/AppDomain.cs (hunk 1 above)
Comment (Therzok) [reply mcs/class/corlib/System/AppDomain.cs]: Shouldn't this be an aka.ms link?

**Comment by steveisok** [comment]
@monojenkins backport 2019-12

**Comment by steveisok** [comment]
@monojenkins backport 2019-10

**Pull request merged**